tmol/pose/pose_stack_builder.py

```diff
@@ -0,0 +1,1964 @@
+import toolz
+import copy
+
+import itertools
+
+import numpy
+import torch
+import pandas
+import sparse
+import scipy.sparse.csgraph as csgraph
+
+from typing import List, Tuple, Optional
+
+from tmol.types.array import NDArray
+from tmol.types.torch import Tensor
+
+from tmol.chemical.constants import MAX_SIG_BOND_SEPARATION
+from tmol.chemical.restypes import (
+    RefinedResidueType,
+    Residue,
+    find_simple_polymeric_connections,
+)
+from tmol.pose.packed_block_types import PackedBlockTypes, residue_types_from_residues
+from tmol.pose.pose_stack import PoseStack
+
+# from tmol.system.datatypes import connection_metadata_dtype
+from tmol.utility.tensor.common_operations import (
+    exclusive_cumsum1d,
+    exclusive_cumsum2d,
+    exclusive_cumsum2d_and_totals,
+    stretch,
+    stretch2,
+)
+from tmol.types.functional import validate_args
+
+
+class PoseStackBuilder:
```

Comment:
this looks good.  BUT I would think about what interfaces we want to support as "public" moving forward.

I believe the first two (one_structure...) are used for testing but perhaps don't make a ton of sense to make as public interfaces?  Perhaps better to prepend with '_'?